codegen/validation.go

```diff
@@ -221,12 +221,13 @@ func recurseValidationCode(att *expr.AttributeExpr, attCtx *AttributeContext, re
 		ctx := attCtx.Dup()
 		ctx.Pointer = false
 		val := recurseValidationCode(a.ElemType, ctx, true, "e", context+"[*]", seen).String()
+
```

Comment:
Hello James, thank you for the PR! this check was supposed to ensure that Goa would only generate a validation method when necessary. Seems that we would need to understand why `val` is the empty string in this case when it shouldn't be rather than removing the test?

Reply:
Thanks for the quick response Raphael. I do agree with you that we need to understand why `val` is empty.